crates/router/src/core/payments.rs

```diff
@@ -3690,60 +3698,202 @@ pub async fn decide_multiplex_connector_for_normal_or_recurring_payment<F: Clone
+                                token_exp_month: token_exp_data.token_exp_month,
+                                token_exp_year: token_exp_data.token_exp_year,
+                            },
                         ));
+                    let connector_data = filtered_nt_supported_connectors
+                        .first()
+                        .ok_or(errors::ApiErrorResponse::IncorrectPaymentMethodConfiguration)
+                        .attach_printable(
+                            "no eligible connector found for token-based MIT payment",
+                        )?;
+
+                    let connector_choice =
+                        Some((connector_data.clone(), mandate_reference_id.clone()));
+                    let (chosen_connector_data, mandate_reference_id) = connector_choice
+                        .get_required_value("connector_choice")
+                        .change_context(
+                            errors::ApiErrorResponse::IncorrectPaymentMethodConfiguration,
+                        )
+                        .attach_printable(
+                            "no eligible connector found for token-based MIT payment",
+                        )?;
 
-                    connector_choice = Some((connector_data, mandate_reference_id.clone()));
-                    break;
-                } else if connector_mandate_details
-                    .clone()
-                    .map(|connector_mandate_details| {
-                        connector_mandate_details.contains_key(merchant_connector_id)
-                    })
-                    .unwrap_or(false)
-                {
-                    if let Some(merchant_connector_id) =
-                        connector_data.merchant_connector_id.as_ref()
-                    {
-                        if let Some(mandate_reference_record) = connector_mandate_details.clone()
+                    routing_data.routed_through =
+                        Some(chosen_connector_data.connector_name.to_string());
+
+                    routing_data
+                        .merchant_connector_id
+                        .clone_from(&chosen_connector_data.merchant_connector_id);
+
+                    payment_data.mandate_id = Some(payments_api::MandateIds {
+                        mandate_id: None,
+                        mandate_reference_id,
+                    });
+
+                    Ok(ConnectorCallType::PreDetermined(chosen_connector_data))
+                }
+                None => {
+                    decide_connector_for_normal_or_recurring_payment(
+                        state,
+                        payment_data,
+                        routing_data,
+                        connectors,
+                        is_connector_agnostic_mit_enabled,
+                        &payment_method_info,
+                    )
+                    .await
+                }
+            }
+        }
+        (
+            None,
```

Comment:
Add comments explaining the None, Some(true), and it other branches of match also add comments when there is None and Boolean.